core/shared/src/main/scala-2.13/fs2/ChunkPlatform.scala

```diff
@@ -0,0 +1,41 @@
+package fs2
+
+import scala.collection.immutable.ArraySeq
+import scala.collection.{immutable, mutable}
+import scala.reflect.ClassTag
+
+trait ChunkPlatform[+O] { self: Chunk[O] =>
+
+  def toArraySeq[O2 >: O: ClassTag]: ArraySeq[O2] = {
+    val array: Array[O2] = new Array[O2](size)
+    copyToArray(array)
+    ArraySeq.unsafeWrapArray[O2](array)
+  }
+
+  def toArraySeqUntagged: ArraySeq[O] =
+    self match {
+      case knownType: Chunk.KnownElementType[O] => toArraySeq[O](knownType.elementClassTag)
+      case _ =>
+        val buf = ArraySeq.untagged.newBuilder[O]
+        buf.sizeHint(size)
+        var i = 0
+        while (i < size) {
+          buf += apply(i)
+          i += 1
+        }
+        buf.result()
+    }
+
+}
+
+trait ChunkCompanionPlatform { self: Chunk.type =>
+
+  // The array casts below are safe because the array constructor will check for primative vs boxed arrays
```

Comment:
This comment replaces the now-deleted comment [here](https://github.com/functional-streams-for-scala/fs2/blob/v2.4.2/core/shared/src/main/scala/fs2/Chunk.scala#L599) in `Chunk`.